### PR TITLE
Support advancements

### LINKED ISSUE #1: Fabric - Advancements
I made a fork that adds support for advancements via "[Iceberg](https://www.curseforge.com/minecraft/mc-mods/iceberg-fabric)", an external dependency:
https://github.com/unilock/yep-paper/tree/fabric-fix

...But it would probably be best to keep yep standalone, for ease of updating, etc.

Looks like Iceberg uses a mixin ([PlayerAdvancementsMixin](https://github.com/AHilyard/Iceberg/blob/1.19-fabric/src/main/java/com/anthonyhilyard/iceberg/mixin/PlayerAdvancementsMixin.java)) to inject an event ([CriterionCallback](https://github.com/AHilyard/Iceberg/blob/1.19-fabric/src/main/java/com/anthonyhilyard/iceberg/events/CriterionCallback.java)) that's fired whenever an advancement is rewarded. (note that Iceberg uses Mojmaps)
I'd reimplement it myself if I knew enough Java to avoid license violations (from blatant copy/pasting).

What do you think?

### PR DESCRIPTION
**Pull request by unilock**
It works!!

Note that I changed `onEntityDeath()` (previously `onPlayerDeath()`) and `sendMessage()` to use `player.getName()` instead of `player.getDisplayName()`, since VelocityDiscord can only support "real names" and not e.g. nicknames.
(...unless?)
<strike>EDIT: I must've broken something, because now player names are showing up as "Realname ~Nickname" :v</strike>
Fixed. Also switched `ALLOW_DEATH` to `AFTER_DEATH` to not send the (cancelled) death message when a Totem of Undying is used.
*Also,* I made it such that advancements are bridged only when the *entire* advancement is granted rather than when only a *criterion* has been achieved - for example, the advancement granted after every Nether biome has been explored is now only mirrored to Discord once *every* biome has been explored, rather than every time an *individual biome* is explored.

Fixes #1.
Partially fixes https://github.com/fooooooooooooooo/VelocityDiscord/issues/4.

### CODE REVIEW

**Comment by fooooooooooooooo** [comment]
is this ready to merge?

**Comment by unilock** [comment]
Should be, yeah.

File: src/main/resources/fabric.mod.json [hunk 1]
```diff
@@ -27,9 +27,9 @@
     ],
 
     "depends": {
-        "fabricloader": ">=0.14.9",
+        "fabricloader": ">=0.14.11",
         "fabric-api": "*",
-        "minecraft": "~1.19",
+        "minecraft": ">=1.19.2",
```
Comment (unilock) [inline src/main/resources/fabric.mod.json:32]: `">="` should be `"~"`, oops :v

(`">=1.19.2"` includes *every* version above or equal to 1.19.2 (incl. 1.20), while `"~1.19.2"` only includes 1.19's *minor versions* above or equal to 1.19.2 (i.e. 1.19.2, 1.19.3, etc., but NOT 1.20 or above)).

Comment (unilock) [reply src/main/resources/fabric.mod.json:32]: Also, leaving it as `"~1.19"` probably would've been fine...

**Pull request merged**